examples/config.py

```diff
@@ -0,0 +1,148 @@
+#   products derived from this software without specific prior
+#   written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""This script allows a user to launch a chain of NFs from a JSON
+config file passed in as an argument when running the script"""
+
+import sys
+import json
+import os
+import shlex
+import time
+from signal import signal, SIGINT
+from subprocess import Popen
+from datetime import datetime
+
+def handler(signal_received, frame):
+    """Handles cleanup on user shutdown"""
+    # pylint: disable=unused-argument
+    for lf in log_files:
+        lf.close()
+    print("\nExiting...")
+    sys.exit(0)
+
+def get_config():
+    """Gets path to JSON config file"""
+    if len(sys.argv) < 2:
+        print("Error: Specify config file")
+        sys.exit(0)
+
+    file_name = sys.argv[1]
+    if os.path.exists(file_name):
+        return file_name
+    print("Error: No config file found")
+    sys.exit(0)
+
+def on_failure(error_output):
+    """Handles shutdown on error"""
+    for lf in log_files:
+        lf.close()
+    for n in nf_list:
+        try:
+            os.system("sudo pkill " + n)
+        except OSError:
+            pass
+    print(error_output)
+    print("Error occurred. Exiting...")
+    sys.exit(1)
+
+def running_services():
+    """Checks running NFs"""
+    while 1:
+        for pl in procs_list:
+            ret_code = pl.poll()
+            if ret_code is not None:
+                out, err = pl.communicate()
+                on_failure(out + err)
+                break
+            time.sleep(.1)
+            continue
+
+procs_list = []
+nf_list = []
+cmds_list = []
+log_files = []
+if __name__ == '__main__':
+    signal(SIGINT, handler)
+    if os.path.basename(os.getcwd()) != "examples":
+        print("Error: Run script from within /examples folder")
+        sys.exit(1)
+
+    if len(sys.argv) < 3:
+        time_obj = datetime.now().time()
+        log_dir = time_obj.strftime("%H:%M:%S")
+    else:
+        log_dir = sys.argv[2]
+    try:
+        os.mkdir(log_dir)
+    except OSError:
+        on_failure("Creation of directory %s failed" % (dir))
+
+    config_file = get_config()
+
+    with open(config_file) as f:
+        data = json.load(f)
```

Comment:
If the json is improperly formatted, we get an error here:
```python
Traceback (most recent call last):
  File "config.py", line 119, in <module>
    data = json.load(f)
  File "/usr/lib/python3.6/json/__init__.py", line 299, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 15 column 1 (char 311)
```

Not sure if this is something you really want to or need to fix, since the user should be using correctly formatted json, but something to consider nonetheless. That error was caused by including an extra comma after the closing `]` for speed_tester

Reply:
We definitely want it to print an error statement if the JSON file has incorrect syntax. It used to do this, but something weird happened when I added the log files, so I'll look into this